### ISSUE TITLE
Disposing container depending of its scene

### ISSUE THREAD

**Issue by MexicanMan** [gustavopsantos/Reflex#12]
Hello! Great framework, but faced the following situation when used scene context with Unity multiscene workflow:
1. Place scene context on scene `A` that is loaded first. That leads us to the initialization of scene scoped container
2. Load scene `B` in the additive mode. Here we can use container from scene `A`.
3. Unload only scene `B`. Here we are disposing container that was initialized and scoped on scene `A`.

As a result any objects created later on scene `A` cannot be resolved anymore with the old container as it had been already disposed. And any additionally loaded scenes after all of that cannot be resolved too.

I don't know if that was supposed to work in a such way and if Reflex should support multiscene workflow at all. But it seems that with the small fix it could be changed to work with multiscene correctly. In [_SceneInjector.cs_](https://github.com/gustavopsantos/Reflex/blob/main/Assets/Reflex/Scripts/Injectors/SceneInjector.cs) we need to change `DisposeScopedContainer` and add the loaded scene to the closure:
```csharp
internal static void Inject(Scene scene, ContainerStack containerStack)
{
	if (scene.TryFindAtRootObjects<Context>(out var sceneContext))
	{
		var container = containerStack.PushNew();
		sceneContext.InstallBindings(container);
		container.InstantiateNonLazySingletons();

		Scene loadedScene = scene;
		void DisposeScopedContainer(Scene unloadingScene)
		{
			if (unloadingScene == loadedScene)
			{
				containerStack.Pop().Dispose();
				SceneManager.sceneUnloaded -= DisposeScopedContainer;
			}
		}

		SceneManager.sceneUnloaded += DisposeScopedContainer;
	}
	
	foreach (var monoBehaviour in GetEveryMonoBehaviourAtScene(scene))
	{
		MonoInjector.Inject(monoBehaviour, containerStack);
	}
}
``` 

So, the container will now only be disposed if its scene is unloaded. And any further additionally loaded scenes could still use previously loaded containers.

**Comment by gustavopsantos** [comment]
Hello @MexicanMan, thanks for pointing this out, Im gonna check it, but yeah, reflex should work with multiscene workflow as well. I'll have time to check this on weekend, Thanks again!

**Comment by gustavopsantos** [comment]
@MexicanMan I pushed a fix for this case you described, but I found another issue, in case you push first, scene A and then scene B, and unload them in the oposite order, the stack will not work in the intended way.
So I need to change the stack to a graph or hierarchy container where we can push and remove objects also in the middle of the tree. Im already working on this at develop branch, soon it will reach main branch ;).
Thanks again @MexicanMan
